Asignment7/baseball_stats.sql

```diff
@@ -0,0 +1,485 @@
+    JOIN s_team_rolling_OBP
+    ON s_baseball_game_stats.away_team_id = s_team_rolling_OBP.team_id and s_baseball_game_stats.game_id = s_team_rolling_OBP.game_id
+    JOIN s_starting_pitcher_rolling_WHIP
+    ON s_baseball_game_stats.away_team_id = s_starting_pitcher_rolling_WHIP.team_id and s_baseball_game_stats.game_id = s_starting_pitcher_rolling_WHIP.game_id
+    JOIN s_away_team_HCP
+    ON s_baseball_game_stats.away_team_id = s_away_team_HCP.away_team_id and s_baseball_game_stats.game_id = s_away_team_HCP.game_id
+    JOIN s_starting_pitcher_K_per_9inn
+    ON s_baseball_game_stats.away_team_id = s_starting_pitcher_K_per_9inn.team_id and s_baseball_game_stats.game_id = s_starting_pitcher_K_per_9inn.game_id
+    JOIN s_starting_pitcher_BB_per_9inn
+    ON s_baseball_game_stats.away_team_id = s_starting_pitcher_BB_per_9inn.team_id and s_baseball_game_stats.game_id = s_starting_pitcher_BB_per_9inn.game_id
+    JOIN s_starting_pitcher_H_per_9inn
+    ON s_baseball_game_stats.away_team_id = s_starting_pitcher_H_per_9inn.team_id and s_baseball_game_stats.game_id = s_starting_pitcher_H_per_9inn.game_id;
+
+
+-- combining all team features to a final table
+
+DROP TABLE IF EXISTS baseball_game_stats;
+CREATE TABLE baseball_game_stats
+
+    SELECT baseball_home_game_stats.game_id,
+           home_team_id,
+           away_team_id,
+           local_date,
+           stadium_id,
+           temperature_in_degrees,
+           Wind_in_mph,
+           wind_direction,
+           home_team_batting_avg,
+           away_team_batting_avg,
+           home_team_home_runs,
+           away_team_home_runs,
+           home_team_pitching_avg,
+           away_team_pitching_avg,
+           home_team_BABIP,
+           away_team_BABIP,
+           home_team_BAIP,
+           away_team_BAIP,
+           home_team_striketowalk,
+           away_team_striketowalk,
+           home_team_walktostrike,
+           away_team_walktostrike,
+           home_team_OBP,
+           away_team_OBP,
+           home_team_HCP,
+           away_team_HCP,
+           home_team_startingpitcher_WHIP,
+           away_team_startingpitcher_WHIP,
+           home_team_startingpitcher_K_per_9_inn,
+           away_team_startingpitcher_K_per_9_inn,
+           home_team_startingpitcher_BB_per_9_inn,
+           away_team_startingpitcher_BB_per_9_inn,
+           home_team_startingpitcher_H_per_9_inn,
+           away_team_startingpitcher_H_per_9_inn,
+           baseball_home_game_stats.winner
+           FROM baseball_home_game_stats
+    JOIN baseball_away_game_stats
+    ON baseball_home_game_stats.game_id = baseball_away_game_stats.game_id;
+
+-- Updating the winner column by replacing H and A with 1 and 0
+
+UPDATE baseball_game_stats
+SET winner = REPLACE(winner,'A','0');
+UPDATE baseball_game_stats
+SET winner = REPLACE(winner,'H','1');
+
+
+SELECT * FROM baseball_game_stats LIMIT 0,20;
```

Comment:
You should take a look into your JOINS as the final table only has game_ids which have values and dropped other game ids. Take a look:
`SELECT * FROM baseball_game_stats
ORDER BY game_id;`